Feat(eos_designs): Add deprecation warning for direct usage of eos_cli_config_gen keys when using eos_designs. (#5775)
aristanetworks/avd: docs/warn-eos-cli-config-keys-usage-in-eos-designs.md

<!--
  ~ Copyright (c) 2023-2025 Arista Networks, Inc.
  ~ Use of this source code is governed by the Apache License 2.0
  ~ that can be found in the LICENSE file.
  -->

> # Of using `eos_cli_config_gen` native keys when running `eos_designs`

Author: if we merge #5780 before - this file should be added to the not_in_nav section (maybe will do it here only for this file)

Reviewer: I doubt it, so please add this in that PR after this one has merged.

Author: I don't doubt it as I have seen it :)

## Context

> For several AVD major versions, it has been possible to use variables from `eos_cli_config_gen` schema in conjunction with variables of `eos_designs`. When `eos_designs`

Reviewer: More like "Since the beginning of time" :) (find a better phrasing, but it was really since 1.0

Reviewer: ```suggestion
For several AVD major versions, it has been possible to use variables from the `eos_cli_config_gen` data model in conjunction with variables from the `eos_designs` data model. When the `eos_designs`
```

> role is run, the `eos_cli_config_gen` keys are ignored and they are injected at `eos_cli_config_gen` role runtime.

Reviewer: ```suggestion
role runs, the `eos_cli_config_gen` keys are ignored but they will be read later by the `eos_cli_config_gen` role.
```

> While a good escape hatch, this has led to confusion among users. Especially when new features introduced in `eos_designs` ended up shadowing `eos_cli_config_gen`

Reviewer: ```suggestion
While this behavior could serve as a good escape hatch, it has led to confusion among users. Especially when new features introduced in `eos_designs` ended up shadowing the inputs intended for `eos_cli_config_gen`,
```

> variables, making some of the confiiguration change sometimes during minor releases. This behavior is well documented and described in the `eos_designs` how-to on

Reviewer: ```suggestion
sometimes leading to configuration changes after a minor releases upgrade. This behavior is well documented and described in the `eos_designs` how-to on
```

[custom_structured_configuration](../ansible_collections/arista/avd/roles/eos_designs/docs/how-to/custom-structured-configuration.md) which is the recommended way of
using `eos_cli_config_gen` variables within `eos_designs`. However the direct usage of `eos_cli_config_gen` variables still works.

## Changes

### AVD 5.6

> Starting AVD 5.6, `eos_designs` role emits deprecation warning identifying the native `eos_cli_config_gen` keys being used.

Reviewer: ```suggestion
Starting AVD 5.6, the `eos_designs` role emits deprecation warning identifying the native `eos_cli_config_gen` keys being used.
```
